controllers/extensions/addon_controller_stages.go

```diff
@@ -1227,6 +1236,15 @@ func validateVersion(annotations, kbVersion string) (bool, error) {
 	return validate, nil
 }
 
+func checkAddonSpec(addon *extensionsv1alpha1.Addon) error {
+	if addon.Spec.Type == "Helm" {
```

Comment:
if addon.Spec.Type == HelmType

should use a constant variable instead of raw string